docs/new-editor.md

```diff
@@ -0,0 +1,163 @@
+[`workspace/didChangeWatchedFiles`][] notifications. The server expects the
+client to send notifications for changes to files matching the following
+patterns
+
+* `.metals/buildinfo/**/*.properties`: build metadata to enable goto definition
+  for the classpath, completions with the presentation compiler and refactoring
+  with Scalafix.
+* `**/*.semanticdb`: artifacts produced by the semanticdb-scalac compiler plugin
+  during batch compilation in the build. See
+  [here](https://github.com/scalameta/scalameta/blob/master/semanticdb/semanticdb3/semanticdb3.md)
+  to learn more about SemanticDB. These files are required for goto definition,
+  find references, hover and Scalafix to work.
+
+See the VS Code plugin
+[clientOptions](https://github.com/scalameta/metals/blob/fb166f1d81eb77ebd9c6b3ee95e65fb58a907eec/vscode-extension/src/extension.ts#L44-L54)
+for an example of how file watching is handled in VS Code.
+
+The metals server scans the workspace directory for these file patterns on
+startup regardless of file watching notifications. This means goto definition
+and other functionality may work to begin with, but quickly degrade for longer
+running sessions.
+
+NOTE. The metals server does not use `DidChangeWatchedFilesRegistrationOptions`
+to register these particular patterns, consider contributing to [#255][] if you
+are affected by this issue.
+
+## Configuration
+
+Metals uses
+[`workspace/didChangeConfiguration`](https://microsoft.github.io/language-server-protocol/specification#workspace_didChangeWatchedFiles)
+notifications to allow end-users to control behavior of the server. The editor
+client is expected to send a `workspace/didChangeConfiguration` notification
+containing user configuration right after the `initialized` notification.
+
+A full list of supported configuration options can be found in the VS Code
+plugin
+[package.json](https://github.com/scalameta/metals/blob/master/vscode-extension/package.json)
+manifest. Some configuration options are required by the server while others
+should be handled by the client.
+
+Server side configuration options include settings to enable
+experimental/unstable features such as completions with the presentation
+compiler. Observe that even if metals registers the capability for completions
+during the `initialize` handshake, the end-user must opt-into enabling
+completions with the configuration setting `scalac.completions.enabled=true`. If
+completions are disabled in the configuration
+(`scalac.completions.enabled=false`, default value), then metals responds with
+an empty list of completion suggestions.
+
+Client side configuration options include
+
+* `serverVersion: String`: while metals is still under active development, it is
+  recommended to allow end-users to easily configure the version of the metals
+  server.
+* `scalafmt.onSave: Boolean`: if true, runs scalafmt on
+  [`textDocument/willSaveWaitUntil`][] client requests.
```

Comment:
I would phrase this paragraph differently.

> Clients are also encouraged to implement this setting:

> * `serverVersion: String`: allow end-users to easily configure the version of the metals. While metals is still under active development, this option is great for flexibly switching to a custom server version.

> **Note**: we recommend to implement the `scalafmt.onSave: Boolean` setting only if the editor doesn't already have a more general mechanism for hanlding formatting on save. For example, thi setting is not included in the Visual Studio Code extension, becuase that particular editor has a dedicated API for it.

Reply:
Thanks for the rephrase, I like it 👍